tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -126,12 +128,13 @@ export const JobSelection = (props, context) => {
                 </span>
               );
             }}
-            jobEntryBuilder={(job, department) => {
+            jobEntryBuilder={(jobName, job, department) => {
               return (
                 <JobEntry
-                  key={job.name}
-                  job={job}
-                  department={department}
+                  key={jobName}
+                  jobName={jobName}
+                  job={job as Job}
+                  department={department as Department}
```

Comment:
Yikes. If you have to forcibly cast types, it's a signal that this abstraction is way too complicated and rigid. You could refactor `DepartmentPane` to use a generic type for Job and Department so that jobEntryBuilder and titleSubtextBuilder would be aware of implemented job types.

Reply:
Okay, TS's syntax for generics is confusing as all hell, unfortunately, so I'll look at this when I'm not as tired.